### PR TITLE
use of minimist to introduce easy options

### PR DESCRIPTION
**Pull request by viqueen**

### CODE REVIEW

**Comment by mgol** [comment]
Thanks for the PR. Could you rebase on `master` and make sure `npm test` passes? Your changes are written in a different code style than the rest of the code base. I haven't had linting enabled for the `bin` directory, that's now fixed.

Also, please describe the features added in README.

File: Gruntfile.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ module.exports = function (grunt) {
             all: {
                 src: [
                     '*.js',
-                    'bin',
+                    'bin/npm-bump',
```
Comment (mgol) [inline Gruntfile.js]: What's the reason for this change?

File: bin/npm-bump [hunk 2]
```diff
@@ -2,14 +2,34 @@
...
+    suffix: opts.suffix,
+  })(opts._[0] || opts.type);
 } catch (error) {
-    if (error.name === 'UsageError') {
-        console.error(error.message);
-    } else {
-        throw error;
-    }
+  if (error.name === 'UsageError') {
+    console.log(error.message);
```
Comment (mgol) [inline bin/npm-bump]: `console.error` seems like a better choice here.

File: Gruntfile.js (hunk 1 above)
Comment (viqueen) [inline Gruntfile.js]: the linter didn't pick up the file, perhaps in my environment only.

**Comment by viqueen** [comment]
haven't had time to update the readme file, will wrap it up sometime on Friday

**Comment by mgol** [comment]
@viqueen Ping :)

**Comment by viqueen** [comment]
I was away for the past month and got distracted with other work things before that. 
I am adding this PR to my backlog this week :)

**Comment by viqueen** [comment]
updated the readme, not sure why the build is failing

**Comment by mgol** [comment]
Sorry for the delay. I think this was this build: https://ci.appveyor.com/project/mgol/npm-bump/build/115/job/7ouxwk1i5di1fg98. It says:
```
>> Error: Cannot find module 'rewire'
```

File: Gruntfile.js (hunk 1 above)
Comment (mgol) [inline Gruntfile.js]: You're right, it's because if you enter a directory here it only matches files with the `.js` extension.

File: bin/npm-bump [hunk 3]
```diff
@@ -2,14 +2,34 @@
 
 'use strict';
 
-const args = process.argv.slice(2);
+const minimist = require('minimist');
+const npmbump = require('../lib/cli');
+
+const opts = minimist(process.argv.slice(2), {
+  alias: {
+    r: 'remote',
```
Comment (mgol) [inline bin/npm-bump]: I think it has incorrect indentation.

**Comment by mgol** [comment]
I have some minor remarks but overall it looks good! Please rebase it over latest master, you'll get a more strict ESLint config that enforces correct indentation, for instance.

I'd love some tests if I only had test infrastructure set up here. 😄 (that's the reason the version is still `0.0.x`) Before that I guess we have to go without them.

File: bin/npm-bump (hunk 3 above)
Comment (viqueen) [inline bin/npm-bump]: I will try and make some time this week for this. crazy busy lately

**Comment by mgol** [comment]
Sure, no worries.

-- 
Michał Gołębiowski

**Comment by mgol** [comment]
Any updates? :)

**Comment by mgol** [comment]
@viqueen Are you still interested in finalizing this PR?

**Comment by viqueen** [comment]
Thanks a lot, apologies I really got side tracked and barely checked my repos/notifications on github.
